features/enlinkd/config/src/main/java/org/opennms/netmgt/config/EnhancedLinkdConfigFactory.java

```diff
@@ -107,7 +106,7 @@ public void reload() throws IOException {
                 reloadXML(stream);
             } finally {
                 if (stream != null) {
```

Comment:
I think we can remove the _if (stream != null) {_
IOUtils.close(stream) should be null safe:
https://commons.apache.org/proper/commons-io/apidocs/org/apache/commons/io/IOUtils.html#close-java.io.Closeable...-

Reply:
Or even better, convert it to the newer try/catch format:

```java
try(final InputStream stream = new FileInputStream(cfgFile)) {
  reloadXML(stream);
}
```

Reply:
/**
     * <p>reload</p>
     *
     * @throws java.io.IOException if any.
     */
    public void reload() throws IOException {
        getWriteLock().lock();
        try {
            final File cfgFile = ConfigFileConstants.getFile(ConfigFileConstants.ENLINKD_CONFIG_FILE_NAME);
            LOG.debug("init: config file path: {}", cfgFile.getPath());
            try (final InputStream stream = new FileInputStream(cfgFile)){
                try(final Reader reader = new InputStreamReader(stream)) {
                    m_config = JaxbUtils.unmarshal(EnlinkdConfiguration.class, reader);
                } finally {
                    IOUtils.close(stream);
                }
            }
            LOG.debug("init: finished loading config file: {}", cfgFile.getPath());
        } finally {
            getWriteLock().unlock();
        }
    }

Reply:
commit bc4349e5486eaad878a7641896d139bf7ca2e171 (HEAD -> jira/ar/NMS-14433, origin/jira/ar/NMS-14433)
Author: Antonio <rssntn67@yahoo.it>
Date:   Thu Jun 23 07:59:08 2022 +0200

    NMS-14433 Enlinkd Clean Config Classes

    Removed unused contructor from EnhancedLinkdConfigFactory

commit e86a231047d0a0a5db4e5d652dffa4d104e541e9
Author: Antonio <rssntn67@yahoo.it>
Date:   Thu Jun 23 07:39:29 2022 +0200

    NMS-14433 Enlinkd Clean Config Classes

    Removed method reloadXML, put all into reload
    Resolve conversation with patrick and ben